From: "Jason A. Donenfeld" <Jason@zx2c4.com>
Date: Thu, 15 Aug 2019 06:37:10 +0000
Subject: compat: fix on debian 4.9.168 in sloppy manner

Ben backported siphash but forgot to backport hsiphash,
which goes into the same .h and .o files, making a proper
compat layer for it kind of cumbersome and annoying. Since
he plans to backport hsiphash for 4.9.169, this commit is
a stop-gap solution for the Debian package.
---
 src/compat/compat.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/compat/compat.h b/src/compat/compat.h
index 96f3556..0a0aac1 100644
--- a/src/compat/compat.h
+++ b/src/compat/compat.h
@@ -1051,6 +1051,12 @@ static inline void skb_reset_redirect(st
 #endif
 #endif
 
+#if LINUX_VERSION_CODE == KERNEL_VERSION(4, 9, 168)
+#define hsiphash_2u32 siphash_2u32
+#define hsiphash_3u32 siphash_3u32
+#define hsiphash_key_t siphash_key_t
+#endif
+
 /* https://github.com/ClangBuiltLinux/linux/issues/7 */
 #if defined( __clang__) && (!defined(CONFIG_CLANG_VERSION) || CONFIG_CLANG_VERSION < 80000)
 #include <linux/bug.h>
